opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/AppResources.java

```diff
@@ -105,112 +137,202 @@ public Response getApp(@QueryParam("appid") Long appId)
+			}
+			else if (appStat.getPid() == null)
+			{
+				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
+				throw new WebAppException(HttpServletResponse.SC_NOT_FOUND, "appid " + appId
+						+ " (" + appStat.getAppName() + ") is not running.");
 			}
 			else if (System.currentTimeMillis() - appStat.getHeartbeat().getTime() > 20000L)
 			{
 				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
-				throw new WebAppException(ErrorCodes.NO_SUCH_OBJECT, "appid " + appId 
-					+ " (" + appStat.getAppName() + ") is not running (stale heartbeat).");
+				throw new WebAppException(HttpServletResponse.SC_NOT_FOUND, "appid " + appId
+						+ " (" + appStat.getAppName() + ") is not running (stale heartbeat).");
 			}
 			else if (!cli.isPresent())
 			{
 				Integer port = appStat.getEventPort();
 				if (port == null)
-					return ApiHttpUtil.createResponse(new ArrayList<ApiAppEvent>());
+				{
+					return Response.status(HttpServletResponse.SC_OK)
+							.entity(new ArrayList<ApiAppEvent>()).build();
+				}
 				apiEventClient = new ApiEventClient(appId, appStat.getHostname(), port, appStat.getAppName(), appStat.getPid());
 				apiEventClient.connect();
-				LOGGER.debug("Connected to {}:{}", appStat.getHostname(), port);
 				clientConnectionCache.addApiEventClient(apiEventClient, session.getId());
 			}
 			else if (appStat.getPid() != null && appStat.getPid() != cli.get().getPid())
 			{
 				// This means that the app was stopped and restarted since we last checked for events.
 				// Close the old client and open a new one with the correct PID.
 				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
-				
+
 				Integer port = appStat.getEventPort();
 				if (port == null)
-					return ApiHttpUtil.createResponse(new ArrayList<ApiAppEvent>()); // app not running
+				{
+					return Response.status(HttpServletResponse.SC_OK)
+							.entity(new ArrayList<ApiAppEvent>()).build();
+				}
 				apiEventClient = new ApiEventClient(appId, appStat.getHostname(), port, appStat.getAppName(), appStat.getPid());
 				apiEventClient.connect();
-				LOGGER.debug("Connected to {}:{}", appStat.getHostname(), port);
 				clientConnectionCache.addApiEventClient(apiEventClient, session.getId());
 			}
 			if(apiEventClient == null)
 			{
-				throw new WebAppException(ErrorCodes.NO_SUCH_OBJECT, "No API Event Client found or created");
+				throw new WebAppException(HttpServletResponse.SC_NOT_FOUND, "No API Event Client found or created");
 			}
-			return ApiHttpUtil.createResponse(apiEventClient.getNewEvents());
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity(apiEventClient.getNewEvents()).build();
 		}
 		catch(ConnectException ex)
-		{			
+		{
 			throw new WebAppException(ErrorCodes.IO_ERROR,
 					String.format("Cannot connect to %s.", appStat.getAppName()), ex);
 			// NOTE: event client added to user token ONLY if connect succeeds.
```

Comment:
I think this is trying to point out that if the "connect()" call throws an exception then the clientConnectionCache doesn't get anything added to it.  This comment is weirdly worded for where the comment is located.  A comment about what happens when the connect succeeds should probably be at the connect call or somewhere on the happy path.  At this location the comment should be reworded so its clear what didn't happen.

Reply:
I moved the comment to a more appropriate location and added a comment on why the exception may be thrown